### PR TITLE
Implemented get_tangent_vector for SphericalManifold.

### PR DESCRIPTION
**Pull request by luca-heltai**
This implements the Interface introduced by @bangerth in #2305 for SphericalManifold. 

This is needed in PR #2272 to test things on the sphere. Once this is merged, I'll rebase there.

### CODE REVIEW

**Comment by luca-heltai** [comment]
It think there might still be periodicity issues with the `SphericalManifold` (not related to this PR, but influencing the applicability of this PR nonetheless.... I'll come up with a test for this).

**Comment by bangerth** [comment]
For the record, here was the implementation I started (completely untested) before you posted your patch:
```

template <int dim, int spacedim>
DerivativeForm<1,spacedim,spacedim>
SphericalManifold<dim,spacedim>::push_forward_gradient(const Point<spacedim> &spherical_point) const
{
  Assert(spherical_point[0] >=0.0,
         ExcMessage("Negative radius for given point."));
  const double rho = spherical_point[0];
  const double theta = spherical_point[1];

  DerivativeForm<1,spacedim,spacedim> gradient;
  if (rho > 1e-10)
    switch (spacedim)
      {
      case 2:
        gradient[0][0] = cos(theta);
        gradient[0][1] = -rho*sin(theta);
        gradient[1][0] = sin(theta);
        gradient[1][1] = rho*cos(theta);
        break;
      case 3:
      {
        const double phi= spherical_point[2];
        gradient[0][0] = sin(theta)*cos(phi);
        gradient[0][1] = -rho*cos(theta)*sin(phi);
        gradient[0][2] = rho*sin(theta)*cos(phi);
        gradient[1][0] = sin(theta)*sin(phi);
        gradient[1][1] = rho*cos(theta)*sin(phi);
        gradient[1][2] = rho*sin(theta)*cos(phi);
        gradient[2][0] = cos(theta);
        gradient[2][1] = -rho*sin(theta);
        gradient[2][2] = 0;
        break;
      }
      default:
        Assert(false, ExcInternalError());
      }

  return gradient;
}
```

File: doc/news/changes.h [hunk 1]
```diff
@@ -87,6 +87,12 @@ inconvenience this causes.
 <a name="specific"></a>
 <h3>Specific improvements</h3>
 
+<ol> 
+ <li> New: SphericalManifold() now implements the new interface of ChartManifold()
+ for tangent vectors..
```
Comment (bangerth) [inline doc/news/changes.h]: double period.

In fact, I had meant not to write an entry there yet but instead, once we're finished with all of these tangent implementations, just write something about all of it in the "general" section.

File: source/grid/manifold_lib.cc [hunk 2]
```diff
@@ -131,6 +131,47 @@ SphericalManifold<dim,spacedim>::pull_back(const Point<spacedim> &space_point) c
...
+      {
+      case 2:
+        DX[0][0] = cos(theta);
+        DX[0][1] = -rho*sin(theta);
+        DX[1][0] = sin(theta);
+        DX[1][1] = rho*cos(theta);
+        break;
+      case 3:
+      {
+        const double &phi= spherical_point[2];
```
Comment (bangerth) [inline source/grid/manifold_lib.cc]: remove the reference -- let's just take the value.

File: source/grid/manifold_lib.cc [hunk 3]
```diff
@@ -131,6 +131,47 @@ SphericalManifold<dim,spacedim>::pull_back(const Point<spacedim> &space_point) c
...
+
+        DX[1][0] =     sin(theta)*sin(phi);
+        DX[1][1] = rho*cos(theta)*sin(phi);
+        DX[1][2] = rho*sin(theta)*cos(phi);
+
+        DX[2][0] =      cos(theta);
+        DX[2][1] = -rho*sin(theta);
+        DX[2][2] = 0;
+      }
+      break;
```
Comment (bangerth) [inline source/grid/manifold_lib.cc]: In analogy to #2315, let's move the `break` before the closing brace.

File: source/grid/manifold_lib.cc [hunk 4]
```diff
@@ -131,6 +131,47 @@ SphericalManifold<dim,spacedim>::pull_back(const Point<spacedim> &space_point) c
...
+      default:
+        Assert(false, ExcInternalError());
```
Comment (bangerth) [inline source/grid/manifold_lib.cc]: ExcNotImplemented

File: tests/manifold/chart_manifold_07.cc [hunk 5]
```diff
@@ -18,8 +18,8 @@
 
 
 Point<3> periodicity (/*r=*/0,
-                      /*phi=*/2*numbers::PI,
-                      /*z=*/0);
+                            /*phi=*/2*numbers::PI,
+                            /*z=*/0);
```
Comment (bangerth) [inline tests/manifold/chart_manifold_07.cc]: Wasn't this part of my flawed commit? Can you rebase?

**Comment by bangerth** [comment]
Let me know once you think this is ready for merge.

**Comment by luca-heltai** [comment]
This is ready for merge. There are problems elsewhere, but I think they are not related to this. I'll post a picture in the other PR. I removed the entry in `changes.h`

**Pull request merged**